git-extra/git-update

```diff
@@ -0,0 +1,38 @@
+#!/bin/sh
+
+# Compares the currently installed Git for Windows against latest available release.
+# If versions differ, the bit matched installer is downloaded and run when confirmation
+# to do so is given.
+
+git_update () {
+    local yn="$1"
+    case $yn in
+        -\? | --?\?) echo "Usage: git-update [options]";
+        echo "Options:";
+        echo "  -y, --yes       Automatic yes to download and install prompt";
+        return;;
+    esac
+    local bit=$(ls --color=never -d /mingw* | sed 's/\/mingw//')
+    local releases=$(curl --silent https://api.github.com/repos/git-for-windows/git/releases/latest)
+    local latest=$(echo "$releases" | grep '"tag_name": "v' | sed -E 's/.*"tag_name": "v([^"]*).*/\1/')
+    local version=$(git --version | sed "s/git version //")
+    echo "Git for Windows $version (${bit}bit)"
+    if [ "$latest" != "$version" ]
+    then
+        echo "Update $latest is available"
+        local download=$(echo "$releases" | grep '"browser_download_url": "' | grep "$bit\-bit\.exe" | sed -E 's/.*": "([^"]*).*/\1/')
+        local filename=$(echo "$download" | sed -E 's/.*\/([^\/]*)$/\1/')
+        case $yn in
+            -[Yy]* | --[Yy]*) yn="y"; echo "Downloading $filename";;
+            *) read -p "Download $filename and install [N/y]? " yn;;
+        esac
+        case $yn in
+            [Yy]* ) curl -# -L -o $filename $download; start $filename; ps | grep bash | awk '{print "kill -9 " $2 ";" }' | sh;;
```

Comment:
You are downloading the installer into the current directory. I would create a tempfile like this...
```
installer=$(mktemp -t "gfw-install-XXXXXXX.exe")
```
... and then download the file to this location and `start` it.